src/VisualStudio/Core/Def/LanguageService/AbstractPackage`2.cs

```diff
@@ -58,32 +58,32 @@ private async Task PackageInitializationMainThreadAsync(IProgress<ServiceProgres
             RegisterEditorFactory(editorFactory);
         }
 
-        RegisterLanguageService(typeof(TLanguageService), async cancellationToken =>
-        {
-            // Ensure we're on the BG when creating the language service.
-            await TaskScheduler.Default;
-
-            // Create the language service, tell it to set itself up, then store it in a field
-            // so we can notify it that it's time to clean up.
-            _languageService = CreateLanguageService();
-            await _languageService.SetupAsync(cancellationToken).ConfigureAwait(false);
-
-            return _languageService.ComAggregate!;
-        });
-
         var miscellaneousFilesWorkspace = this.ComponentModel.GetService<MiscellaneousFilesWorkspace>();
 
         // awaiting an IVsTask guarantees to return on the captured context
         await shell.LoadPackageAsync(Guids.RoslynPackageId);
 
-        RegisterMiscellaneousFilesWorkspaceInformation(miscellaneousFilesWorkspace);
+        packageRegistrationTasks.AddTask(
+             isMainThreadTask: false,
+             task: (IProgress<ServiceProgressData> progress, PackageRegistrationTasks packageRegistrationTasks, CancellationToken cancellationToken) =>
+             {
+                 RegisterLanguageService(typeof(TLanguageService), async cancellationToken =>
+                 {
+                     // Ensure we're on the BG when creating the language service.
+                     await TaskScheduler.Default;
```

Comment:
We already said isMainThreadtask: false, why saying this a second time?

Reply:
That's the callback when the service is created, I don't know if we can guarantee what context we are on when that happens.

Reply:
Oh right of course. That'd be something we should follow up with -- hopefully we have some way to say we can be created on a background thread -- it'd be silly for the service system to bounce to the UI thread if we didn't need it.